backend/de.metas.manufacturing/src/main/java/org/eevolution/api/impl/PPOrderDAO.java

```diff
@@ -84,6 +87,18 @@ public Stream<I_PP_Order> streamManufacturingOrders(@NonNull final Manufacturing
 		return toSqlQueryBuilder(query).create().iterateAndStream();
 	}
 
+	@Override
+	public int getLastSeqNoPerOrderDate(@NonNull final I_PP_Order ppOrder)
+	{
+		final int lastSeqNo = Services.get(IQueryBL.class)
+				.createQueryBuilder(I_PP_Order.class, ppOrder)
+				.addEqualsFilter(I_PP_Order.COLUMN_DateOrdered, ppOrder.getDateOrdered())
+				.create()
+				.aggregate(I_M_ProductPrice.COLUMNNAME_SeqNo, IQuery.Aggregate.MAX, int.class);
+
+		return Math.max(lastSeqNo, 0) + 10;
```

Comment:
we recently introduced de.metas.util.lang.SeqNo to help us dealing with these kind of sequence numbers.

pls use something like:
```
return SeqNo.ofInt(lastSeqNo).next()
```

and convert it from SeqNo to int (using SeqNo.toInt()) just when u are setting it to the target field.

wdyt?